### PR TITLE
update k8s deps which include leader election fixes

### LINKED ISSUE #446: Bump k8s to v0.18.12 to get leader election release bug fixes
**Problem**

- Context: https://github.com/knative/eventing/issues/4542
- Similar to https://github.com/knative-sandbox/net-contour/pull/361
- Also remove https://github.com/knative-sandbox/eventing-kafka-broker/blob/master/hack/k8s-client-go.patch

/good-first-issue
/area release

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

Fixes #446

## Proposed Changes

🐛 Leader election transition improvements

**Release Note**

```release-note
Leader election transition improvements
```

**Docs**

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -24,5 +24,4 @@ go_update_deps "$@"
 
 # Patch k8s leader-election fixing graceful release
 # More information: https://github.com/kubernetes/kubernetes/pull/91942
-git apply ${REPO_ROOT_DIR}/hack/k8s-client-go.patch
```
Comment (pierDipi) [inline hack/update-deps.sh:27]: Can you remove this comment as well?

File: go.mod [hunk 2]
```diff
@@ -27,9 +27,11 @@ require (
...
-	k8s.io/client-go => k8s.io/client-go v0.18.8
-	k8s.io/code-generator => k8s.io/code-generator v0.18.8
+	k8s.io/api => k8s.io/api v0.18.12
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.18.12
+	k8s.io/apimachinery => k8s.io/apimachinery v0.18.12
+	k8s.io/client-go => k8s.io/client-go v0.18.12
+	k8s.io/code-generator => k8s.io/code-generator v0.18.12
 )
+
+replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6
```
Comment (pierDipi) [inline go.mod:37]: Can we use the same replace block above?

**Comment by arghya88** [comment]
@pierDipi done the changes.PTAL

**Review by pierDipi — APPROVED** [review]
Thanks, @arghya88!

/lgtm
/approve

**Pull request merged**